### ISSUE TITLE
OF-v2112

### ISSUE THREAD

**Issue by ndkazu** [HenningScheufler/multiDimAMR#7]
Hello, I'm trying to update your code for the latest OF release.
During compilation I get the following error:
```
hexRef/hexRefNew.C:64:5: error: ‘meshConstructorTable’ is not a class, namespace, or enumeration
   64 |     meshConstructorTable::iterator hexRefIter =
      |     ^~~~~~~~~~~~~~~~~~~~
hexRef/hexRefNew.C:67:9: error: ‘hexRefIter’ was not declared in this scope; did you mean ‘hexRef’?
   67 |     if (hexRefIter == meshConstructorTablePtr_->end())
```
can you help me solve it, please?
regards,

Kazunobu

**Comment by HenningScheufler** [comment]
They refactor runtime selection macro i will look into if have time but im currently busy with another project. 

You could compare of2106 to of2112 what the they in the runtime selection mechanism and make a pull request

**Comment by albertoceschin** [comment]
Long story short: use "auto" keyword instead of "meshConstructorTable::iterator" and a few others errors that will appear. It worked well for me.

In the development version of v2112 you can find more details. 

Cheers

**Comment by HenningScheufler** [comment]
@albertoceschin can you make a pull request

**Comment by ndkazu** [comment]
@albertoceschin  thank you, it worked! Also, I made a PR.

**Comment by HenningScheufler** [comment]
Fixed in PR  V2012 #8
